depthai_nodes/ml/parsers/lane_detection.py

```diff
@@ -109,32 +109,40 @@ def setClsNumPerLane(self, cls_num_per_lane: int) -> None:
             raise ValueError("Number of points per lane must be an integer.")
         self.cls_num_per_lane = cls_num_per_lane
 
-    def setInputShape(self, input_shape: Tuple[int, int]) -> None:
-        """Set the input shape for the lane detection model.
+    def setInputSize(self, input_size: Tuple[int, int]) -> None:
+        """Set the input size for the lane detection model.
 
-        @param input_shape: Input shape.
-        @type input_shape: Tuple[int, int]
+        @param input_size: Input size (width,height).
+        @type input_size: Tuple[int, int]
         """
-        if not isinstance(input_shape, tuple):
-            raise ValueError("Input shape must be a tuple.")
-        if len(input_shape) != 2:
-            raise ValueError("Input shape must be a tuple of two integers.")
-        if not all(isinstance(size, int) for size in input_shape):
-            raise ValueError("Input shape must be a tuple of integers.")
-        self.input_shape = input_shape
+        if not isinstance(input_size, tuple):
+            raise ValueError("Input size must be a tuple.")
+        if len(input_size) != 2:
+            raise ValueError("Input size must be a tuple of two integers.")
+        if not all(isinstance(size, int) for size in input_size):
+            raise ValueError("Input size must be a tuple of integers.")
+        self.input_size = input_size
 
     def build(
         self,
         head_config: Dict[str, Any],
+        inputs_size: List[List[int]],
```

Comment:
Hm this breaks the signature of the build function defined in the BaseParser. What if we rather have build in BaseParser defined as:
```python
def build(self, head_config: Dict[str, Any], **kwargs: Dict[str, Any]) -> "BaseParser"
```

And then any additional params get passed are are accesed through kwargs? So for parsers that would need "input_size" we get it in build method as:
```python
self.input_size = kwargs.get("input_size", self.input_size) # use default if not present and default would be None and we error out in run() if it stays None
```
Thoughs?

Reply:
This could work.
Alternative option: we extend `head_config` with inputs information and then move the input_size extraction logic from parser_generator to base_parser `build` method. this way we dont introduce new parameters to any functions. But we need to call `super.build()` in all parsers build methods.

Reply:
Actually this seems a nicer way of doing it yeah. One note: This makes build() not abstract anymore - is there any concern because of this? I can't think of any but it is something to be aware of

Reply:
Both options are fine to me! Also, I don't see a reason why we would want to keep the parent build() method abstract.